### PR TITLE
Add an option to enable/disable OS on project level.

### LINKED ISSUE #6477: Enable/disable operating system for projects
**What would you like to be added? / User Story**:

As a KKP admin, I'd like the option to be able to enable/disable operating systems at project scope. Currently, through https://github.com/kubermatic/dashboard/pull/6070, this support was added to configure this at a global level.

An option to configure this setting at a `project` or `data center` level would be more helpful since configuring them globally for all projects/user-groups/customers is not ideal.

### PR DESCRIPTION
**Pull request by ahmadhamzh**
**What this PR does / why we need it**:
Add an option in the admin setting to hide/show operating system when create machine deployment

**Add/Edit Project:**
image

**Which issue(s) this PR fixes**:
Fixes #6477 

**What type of PR is this?**

/kind feature

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Support for enabling/disabling operating systems for machines in user clusters at the project level.
```

```documentation
TBD
```

### CODE REVIEW

**Review by Waseem826 — CHANGES_REQUESTED** [review]
Some feedback. PTAL

**Comment by Waseem826** [comment]
Shouldn't this setting be restricted to Admins? I don't see any such restriction in the code.

File: modules/web/src/app/node-data/component.ts [hunk 1]
```diff
@@ -295,34 +296,21 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
...
-        this.form.get(Controls.UpgradeOnBoot).setValue(autoUpdatesEnabled);
-        this.form.get(Controls.DisableAutoUpdate).setValue(!autoUpdatesEnabled);
-      }
-      if (settings.machineDeploymentOptions.autoUpdatesEnforced) {
-        this.form.get(Controls.UpgradeOnBoot).disable();
-        this.form.get(Controls.DisableAutoUpdate).disable();
+    forkJoin([
+      this._settingsService.adminSettings.pipe(take(1)),
+      this._projectService.selectedProject.pipe(take(1)),
+    ]).subscribe(([settings, project]) => {
```
Comment (Waseem826) [inline modules/web/src/app/node-data/component.ts:302]: We can apply `take(1)` on the `forkJoin`.

```suggestion
    forkJoin([
      this._settingsService.adminSettings,
      this._projectService.selectedProject,
    ]).pipe(take(1)).subscribe(([settings, project]) => {
```

Comment (ahmadhamzh) [reply modules/web/src/app/node-data/component.ts:302]: Unfortunately, that didn't work.

**Review by Waseem826 — APPROVED** [review]
/approve

**Pull request merged**